src/private/TitleBar.cpp

```diff
@@ -501,14 +546,32 @@ void TitleBar::onAutoHideClicked()
         return;
     }
 
+    const bool groupedAutoHide = Config::self().flags() & Config::Flag_AutoHideAsTabGroups;
+    const auto currentDw = m_frame->currentDockWidget();
+
     const auto &dockwidgets = m_frame->dockWidgets();
     for (DockWidgetBase *dw : dockwidgets) {
         if (dw->isOverlayed()) {
             // restore
             MainWindowBase *mainWindow = dw->mainWindow();
-            mainWindow->restoreFromSideBar(dw);
+            auto sideBarGroup = groupedAutoHide ? SideBarGroupRegister::groupForDockWidget(dw) : DockWidgetBase::List();
+            if (sideBarGroup.isEmpty()) {
+                mainWindow->restoreFromSideBar(dw);
+            } else {
+                for (auto it = sideBarGroup.rbegin(); it != sideBarGroup.rend(); ++it) {
+                    DockWidgetBase *member = *it;
+                    auto title = member->title();
+                    title.clear();
```

Comment:
This is a no-op, it's clearing a local QString variable

Reply:
Right, I added the first line to be able to see it in a debugger. The second line to not have it optimized away :-)